tests/components/qwikswitch/test_init.py

```diff
@@ -98,18 +82,45 @@ async def test_sensor_device(hass, aioclient_mock):  # noqa: F811
             }
         }
     }
+    aioclient_mock.get("http://127.0.0.1:2020/&device", json=DEVICES)
+    listen_mock = aioclient_mock.get("http://127.0.0.1:2020/&listen", long_poll=True)
     await async_setup_component(hass, QWIKSWITCH, config)
-
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
     await hass.async_block_till_done()
+
     state_obj = hass.states.get("sensor.ss1")
     assert state_obj.state == "None"
 
+    listen_mock.queue_response(
+        json={"id": "@a00001", "name": "ss1", "type": "rel", "val": "4733800001a00000"},
+    )
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    state_obj = hass.states.get("sensor.ss1")
+    assert state_obj.state == "416"
+
+
+async def test_switch_device(hass, aioclient_mock):
+    """Test a switch device."""
+    config = {
+        "qwikswitch": {
+            "sensors": {"name": "ss1", "id": "@a00001", "channel": 1, "type": "imod"},
```

Comment:
Type should be “rel” afair, “dim” could potentially also be a switch, but mostly relay

Reply:
Maybe use the DEVICES[0] global var for setup of a switch?

Reply:
Type=“rel” and no channel needed

Reply:
thanks. ended up figuring out that i need to change the DEVICES. just committed something that tests the flows of a switch
it requires another small change in aiohttp, mainly that it goes over the registered mock http responses in reverse order so if the value changes, it will use the latest. let's see if it will be accepted

Reply:
will take the challenge to expand to 100% coverage. i believe i am past the most difficult part